nixos/modules/services/web-apps/ackee.nix

```diff
@@ -0,0 +1,135 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ackee;
+  settings = cfg.defaultSettings // cfg.settings;
+in
+{
+  options.services.ackee = {
+    enable = mkEnableOption "the Ackee analytics tool";
+
+    defaultSettings = mkOption {
```

Comment:
You don't need this. Use the `settings` option and within `config` you can set default values with `mkDefault` which the sysadmin is free to override.